src/main/java/no/finn/unleash/util/UnleashConfig.java

```diff
@@ -36,23 +37,23 @@
     private final UnleashSubscriber unleashSubscriber;
 
     private UnleashConfig(
-            URI unleashAPI,
+            @Nullable URI unleashAPI,
             Map<String, String> customHttpHeaders,
             CustomHttpHeadersProvider customHttpHeadersProvider,
-            String appName,
+            @Nullable String appName,
             String environment,
-            String instanceId,
+            @Nullable String instanceId,
             String sdkVersion,
             String backupFile,
-            String projectName,
+            @Nullable String projectName,
```

Comment:
This constructor has checks for all nullable parameters except the `projectName`. I assume it is OK to be nullable (based on current behavior), but wanted to point out the difference.

Reply:
Good point, yes, it's ok to be nullable, since only enterprise has support for projects at the moment

Reply:
Also because there are situations where you might not want to filter on projectName, leaving it out is perfectly fine.